src/stactools/goes_glm/parquet.py

```diff
@@ -200,35 +199,21 @@ def create_asset(
                 base = datetime.fromisoformat(unit[14:]).replace(tzinfo=timezone.utc)
 
                 new_data: List[Optional[datetime]] = []
-                invalid_datetimes = 0
                 for val in data:
-                    if val is None or isnan(val):
-                        # Sometimes the values contain NaN/None values
-                        new_data.append(None)
-                        invalid_datetimes += 1
-                    else:
-                        try:
-                            delta = timedelta(seconds=val)
-                            new_data.append(base + delta)
-                        except TypeError:
-                            raise Exception(
-                                f"An invalid value '{val}' found in variable '{var_name}'"
-                            )
+                    try:
+                        delta = timedelta(seconds=val)
+                        new_data.append(base + delta)
+                    except TypeError:
+                        raise Exception(
+                            f"An invalid value '{val}' found in variable '{var_name}'"
```

Comment:
Although _Unsigned is set to true, this will be thrown for at least one test due to a None value event_time_offset. So either there can still be invalid non-numerical values or something doesn't work quite right yet.